File: frontend/src/types/portfolio.ts
export interface Portfolio {
  id: number
  slug: string
  headline: string | null
  bio: string | null
  email: string | null
  phone: string | null
  location: string | null
  linkedin_url: string | null
  github_url: string | null
  skills: string[] | null
  experience: string[] | null
  education: string[] | null
  projects: string[] | null
  certifications: string[] | null
  achievements: string[] | null
  theme: string
  is_published: boolean
  created_at: string
  updated_at: string
}

export interface PortfolioUpdate {
  headline?: string | null
  bio?: string | null
  email?: string | null
  phone?: string | null
  location?: string | null
  linkedin_url?: string | null
  github_url?: string | null
  skills?: string[] | null
  projects?: string[] | null
  is_published?: boolean
}

File: backend/app/schemas/portfolio.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PortfolioGenerate(BaseModel):
    """Generate or refresh a portfolio from an owned parsed resume version."""

    resume_version_id: int = Field(gt=0)


class PortfolioUpdate(BaseModel):
    """User-editable presentation fields for an existing portfolio."""

    headline: str | None = Field(default=None, max_length=255)
    bio: str | None = None
    email: str | None = Field(default=None, max_length=255)
    phone: str | None = Field(default=None, max_length=50)
    location: str | None = Field(default=None, max_length=255)
    linkedin_url: str | None = Field(default=None, max_length=500)
    github_url: str | None = Field(default=None, max_length=500)
    skills: list[str] | None = None
    projects: list[str] | None = None
    is_published: bool | None = None


class PortfolioResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    slug: str
    headline: str | None
    bio: str | None
    email: str | None
    phone: str | None
    location: str | None
    linkedin_url: str | None
    github_url: str | None
    skills: list | None
    experience: list | None
    education: list | None
    projects: list | None
    certifications: list | None
    achievements: list | None
    theme: str
    is_published: bool
    created_at: datetime
    updated_at: datetime


class PublicPortfolioResponse(PortfolioResponse):
    """Published portfolio response. Deliberately excludes account ownership."""

    pass
